src/nuclide_header.F90

```diff
@@ -383,108 +433,125 @@ end subroutine nuclide_from_hdf5
+              allocate(this % index_fission(PARTIAL_FISSION_MAX))
+              this % has_partial_fission = .true.
+            end if
+          end if
 
-        ! Add contribution to fission cross section
-        if (is_fission(rx % MT)) then
-          this % fissionable = .true.
-          this % fission(j:j+n-1) = this % fission(j:j+n-1) + rx % sigma
-
-          ! Also need to add fission cross sections to absorption
-          this % absorption(j:j+n-1) = this % absorption(j:j+n-1) + rx % sigma
-
-          ! If total fission reaction is present, there's no need to store the
-          ! reaction cross-section since it was copied to this % fission
-          if (rx % MT == N_FISSION) deallocate(rx % sigma)
-
-          ! Keep track of this reaction for easy searching later
-          i_fission = i_fission + 1
-          this % index_fission(i_fission) = i
-          this % n_fission = this % n_fission + 1
-
-          ! <<<<<<<<<<<<<<<<<<<<<<<<<<<< REMOVE THIS <<<<<<<<<<<<<<<<<<<<<<<<<<<
-          ! Before the secondary distribution refactor, when the angle/energy
-          ! distribution was uncorrelated, no angle was actually sampled. With
-          ! the refactor, an angle is always sampled for an uncorrelated
-          ! distribution even when no angle distribution exists in the ACE file
-          ! (isotropic is assumed). To preserve the RNG stream, we explicitly
-          ! mark fission reactions so that we avoid the angle sampling.
-          do k = 1, size(rx % products)
-            if (rx % products(k) % particle == NEUTRON) then
-              do m = 1, size(rx % products(k) % distribution)
-                associate (aedist => rx % products(k) % distribution(m) % obj)
-                  select type (aedist)
-                  type is (UncorrelatedAngleEnergy)
-                    aedist % fission = .true.
-                  end select
-                end associate
+          ! Add contribution to fission cross section
+          if (is_fission(rx % MT)) then
+            this % fissionable = .true.
+            this % sum_xs(t) % fission(j:j+n-1) = this % sum_xs(t) % &
+                 fission(j:j+n-1) + rx % xs(t) % value
+
+            ! Also need to add fission cross sections to absorption
+            this % sum_xs(t) % absorption(j:j+n-1) = this % sum_xs(t) % &
+                 absorption(j:j+n-1) + rx % xs(t) % value
+
+            ! If total fission reaction is present, there's no need to store the
+            ! reaction cross-section since it was copied to this % fission
+            if (rx % MT == N_FISSION) deallocate(rx % xs(t) % value)
+
+            ! Keep track of this reaction for easy searching later
+            if (t == 1) then
+              i_fission = i_fission + 1
+              this % index_fission(i_fission) = i
+              this % n_fission = this % n_fission + 1
+
+              ! <<<<<<<<<<<<<<<<<<<<<<<<<<<< REMOVE THIS <<<<<<<<<<<<<<<<<<<<<<<<<
```

Comment:
This has more to do with *where* mu is sampled rather than whether it is sampled. As is, `sample_fission_neutron` samples `mu` and `phi` even though the proper thing to do is to just let the isotropic angular distribution do its thing. However, this changes the order of the RNG stream and hence affects results. When I did the big secondary distribution refactor, I didn't want to change results to prove that the only thing I was changing was how data was abstracted.